web/html/src/manager/minion/ansible/accordion-path-content.tsx

```diff
@@ -0,0 +1,182 @@
+import * as React from "react";
+import { AnsiblePath } from "./ansible-path-type";
+import Network from "utils/network";
+import { Messages, Utils } from "components/messages";
+import { Loading } from "components/utils/Loading";
+import { AceEditor } from "components/ace-editor";
+
+type PropsType = {
+  path: AnsiblePath;
+};
+
+type StateType = {
+  open: boolean; 
+  content: any;
+  errors: string[];
+  loading: boolean;
+};
+
+function getURL(path: AnsiblePath) {
+  let baseUrl: string;
+  if (path.type === "playbook") {
+    baseUrl = "/rhn/manager/api/systems/details/ansible/discover-playbooks/";
+  }
+  else {
+    baseUrl = "/rhn/manager/api/systems/details/ansible/introspect-inventory/";
+  }
+  return baseUrl + path.id;
+}
+
+interface PlaybookDetails {
+  path: AnsiblePath,
+  fullPath: string,
+  customInventory?: string
+  name: string,
+}
+
+class AccordionPathContent extends React.Component<PropsType, StateType> {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      open: false,
+      content: null,
+      errors: [],
+      loading: false,
+    };
+  }
+
+  onToggle() {
+    const path: AnsiblePath = this.props.path;
+    if (!this.state.open) {
+      if (this.state.content === null) {
+        this.setState({ loading: true });
+        Network.get(getURL(path))
+        .promise.then(blob => {
+          if (blob.success) {
+            const data = blob.data;
+            this.setState({
+              content: this.props.path.type === "playbook" ?
+                this.digestPlaybookPathContent(data)
+                :
+                this.digestInventoryPathContent(data)
+            });
+          }
+          else {
+            this.setState({ errors: blob.messages });
+          }
+          this.setState({ open: true, loading: false });
+        });
+      }
+      else {
+        this.setState({ open: true });
+      }
+    }
+    else {
+      this.setState({ open: false, errors: [] });
+    }
+  }
+
+  digestPlaybookPathContent(blob: any) {
+    const content: Map<string, {}> = blob[this.props.path.path];
```

Comment:
You probably meant a `Record` here, unless `blob[this.props.path.path]` is an actual [`Map`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map)?

Reply:
Actually it is a `Map`, yes :/

```
{
  "/root/ansible-examples/lamp_haproxy": {
		"aws/demo-aws-launch.yml": {
			"fullpath": "/root/ansible-examples/lamp_haproxy/aws/demo-aws-launch.yml"
		},
		"aws/rolling_update.yml": {
			"fullpath": "/root/ansible-examples/lamp_haproxy/aws/rolling_update.yml"
		},
		"aws/site.yml": {
			"fullpath": "/root/ansible-examples/lamp_haproxy/aws/site.yml"
		},
		"provision.yml": {
			"custom_inventory": "/root/ansible-examples/lamp_haproxy/hosts",
			"fullpath": "/root/ansible-examples/lamp_haproxy/provision.yml"
		},
		"rolling_update.yml": {
			"custom_inventory": "/root/ansible-examples/lamp_haproxy/hosts",
			"fullpath": "/root/ansible-examples/lamp_haproxy/rolling_update.yml"
		},
		"site.yml": {
			"custom_inventory": "/root/ansible-examples/lamp_haproxy/hosts",
			"fullpath": "/root/ansible-examples/lamp_haproxy/site.yml"
		}
	}
}
```
and I expect the response to be like the one above because I need to parse properly. Do you still think `Record` makes more sense here? Should I adapt it?

Reply:
If `content` is a `Map` then later accesses into it should probably be with `const playbookObj = content.get(playbookName)` (not `const playbookObj = content[playbookName]`) etc. Or perhaps I don't understand what you mean?